python/revng/internal/cli/_commands/mass_testing/configure.py

```diff
@@ -40,9 +40,8 @@ def run(self, options: Options):
 
         # Shuffle the input files, this avoids having a limited variety of
         # test runs for a partial run
-        if args.seed is not None:
-            seed = args.seed
-        else:
+        seed = os.environ.get("MASS_TESTING_CONFIGURE_SEED", args.seed)
```

Comment:
What's the use for these?

Reply:
That's the seed passed to the PRNG, if you use the same seed and the same inputs you'll get the same files (if you use `--sample`) or the same order (due to `shuffle`)

Reply:
Usually arguments take precedence over environment variables, here is the other way around.

Is there a reason to have a non-hardcoded seed?
I understand we want determinism. Do we ever want a *different* set of samples?

Even if we do, setting the seed to `0` might saving us from having to specifiy a seed via environment variables.

Reply:
> Usually arguments take precedence over environment variables, here is the other way around.

Yes, but since these "variables" need to be overridden from outside orchestra it's needed for the environment variable counterpart to have precedence over the command line one

> Is there a reason to have a non-hardcoded seed?

Since the ordering can "hide" bugs (e.g. a crash should happen, but it times out before that) it's beneficial for runs to use a random seed

> Even if we do, setting the seed to 0 might saving us from having to specifiy a seed via environment variables.

That's only needed in "controlled" situations, e.g. when running regression. In general I'm not sure we want that, having variety means that some different bugs will show up (which IMHO is good)

Reply:
I think no one is going to use this other than the mass testing itself.

Let's hardcode a seed and call it a day, if you really want you can use `--seed=$RANDOM` to test more stuff.

Reply:
> if you really want you can use --seed=$RANDOM to test more stuff.

Having only the command-line options is basically useless, if anything then only the env should exist.

> Let's hardcode a seed and call it a day

I don't see any benefit in that, if there's an ordering bug it could get hidden. Or, put another way, there isn't any benefit (from the POV of mass-testing) to **not** shuffle things differently each run.
Note: the seed used for the run is recorded in `meta.yml`, so in any case the run **can** be "reproduced" if needed.

Reply:
OK.

> Usually arguments take precedence over environment variables, here is the other way around.

Please do this though.

Reply:
Dropped the command-line options.